contracts/standard/proxy/ArbitratorVersioningProxy.sol

```diff
@@ -73,27 +83,41 @@ contract ArbitratorVersioningProxy is Arbitrator, VersioningProxy {
     }
 
     /** @notice Computes the cost of appealing to the current `implementation` contract. It is recommended not to increase it often, as it can be highly time and gas consuming for the arbitrated contracts to cope with fee augmentation.
-     *  @param _disputeID The ID of the dispute to be appealed.
+     *  @param _localDisputeID The ID of the dispute to be appealed.
      *  @param _extraData Can be used to give additional info on the dispute to be created.
      *  @return _fee The appeal cost.
      */
-    function appealCost(uint256 _disputeID, bytes _extraData) public view returns(uint256 _fee) {
-        return Arbitrator(implementation).appealCost(disputes[_disputeID].disputeID, _extraData);
+    function appealCost(uint256 _localDisputeID, bytes _extraData) public view returns(uint256 _fee) {
+        return Arbitrator(implementation).appealCost(disputes[_localDisputeID].externalDisputeID, _extraData);
```

Comment:
This won't work.

If the implementation changed we need to return the arbitration cost of the new implementation. Like a mirror of `appeal()`.

Reply:
So we are repeating some statements, I also refactored them out as an internal function.

Reply:
Wait, appealCost can't stay as view when we create a dispute inside it.

Reply:
You are not supposed to create one, you are supposed to return the arbitration cost of creating one.

>If the implementation changed we need to return the arbitration cost of the new implementation. Like a mirror of appeal().